imod/util/expand_repetitions.py

```diff
@@ -15,3 +42,94 @@ def expand_repetitions(
         if newdate < time_max:
             expanded[newdate] = date
     return expanded
+
+
+def resample_timeseries(well_rate: pd.DataFrame, times: list[datetime]) -> pd.DataFrame:
+    """
+    On input, well_rate is a dataframe containing a timeseries for rate for one well
+    while "times" is a list of datetimes.
+    This function creates a new dataframe containing a timeseries defined on the
+    times in "times".
+
+    Parameters
+    ----------
+    well_rate:  pd.DataFrame
+        input timeseries for well
+    times: datetime
+        list of times on which the output datarame should have entries.
+
+    returns:
+    --------
+    a new dataframe containing a timeseries defined on the times in "times".
+    """
+    output_frame = pd.DataFrame(times)
+    output_frame = output_frame.rename(columns={0: "time"})
+    intermediate_df = pd.merge(
+        output_frame,
+        well_rate,
+        how="outer",
+        on="time",
+    ).fillna(method="ffill")
+
+    # the entries before the start of the well timeseries do not have data yet, so we fill them in here
+    if intermediate_df["time"].values[0] < well_rate["time"].values[0]:
+        intermediate_df.loc[intermediate_df["time"] < well_rate["time"][0], "rate"] = (
+            0.0
+        )
+        intermediate_df.loc[intermediate_df["time"] < well_rate["time"][0], "x"] = (
+            well_rate["x"][0]
+        )
+        intermediate_df.loc[intermediate_df["time"] < well_rate["time"][0], "y"] = (
+            well_rate["y"][0]
+        )
+        intermediate_df.loc[intermediate_df["time"] < well_rate["time"][0], "id"] = (
+            well_rate["id"][0]
+        )
+        intermediate_df.loc[
+            intermediate_df["time"] < well_rate["time"][0], "filt_top"
+        ] = well_rate["filt_top"][0]
+        intermediate_df.loc[
+            intermediate_df["time"] < well_rate["time"][0], "filt_bot"
+        ] = well_rate["filt_bot"][0]
```

Comment:
Getting verbose here, I think this should reduce the amount of code here:

```suggestion
    time_before_start_input = intermediate_df["time"] < well_rate["time"][0]
    if time_before_start_input[0]:
        intermediate_df.loc[time_before_start_input, "rate"] = 0.0
        intermediate_df.loc[time_before_start_input, "x"] = well_rate["x"][0]
        intermediate_df.loc[time_before_start_input, "y"] = well_rate["y"][0]
...
#etc
```